### ISSUE TITLE
option(SkipUpsert=true) is not case inSensitive 

### ISSUE THREAD

**Issue by eaugene** [apache/pinot#10832]
Using `option(SkipUpsert=true)` in Query works by including upserted records, while `option(skipUpsert=true)` doesn't work as intended, also is silent and doesn't throw any error when processing query.

Pinot Version:0.11
label: `bug`

**Comment by Jackie-Jiang** [comment]
All query options are case sensitive now, and pinot doesn't throw error when it cannot recognize an option. One solution would be to canonicalize the option keys, and check if the key is within an allowed key list. The problem with this approach is that it makes it hard to extend the option keys because user might want to put some options not understood by pinot, but used in their custom code plugged in

**Comment by eaugene** [comment]
Got it @Jackie-Jiang  . But on the flip side , can we make query options case insensitive?

**Comment by Jackie-Jiang** [comment]
Yes, I think it is a fair ask. What we can do is to always use lower case to store and look up query options. Do you want to raise a PR to add this support?

**Comment by eaugene** [comment]
Sure @Jackie-Jiang  . Please assign to me
